### PR TITLE
chore(dangerjs): check dependency changes

### PR DESCRIPTION
**Pull request by miroslavstastny**
As part of CI, package dependencies are verified - if relevant change is detected (new `package.json` added, new dependency added, version of existing dependency changed in any `package.json`) warning is added to the PR.

### Example

image

### CODE REVIEW

File: dangerfile.ts [hunk 1]
```diff
@@ -37,3 +40,53 @@ if (!hasChangelog) {
...
+    warn(`New package added: ${filepath}. Make sure you have OSS approval before merging!`)
+  })
+
+danger.git.modified_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(async filepath => {
+    const diff = await danger.git.JSONDiffForFile(filepath)
+    if (diff.dependencies) {
+      const before = { ...diff.dependencies.before, ..._.zipObject(diff.dependencies.added) }
+      const after = diff.dependencies.after || {}
```
Comment (layershifter) [inline dangerfile.ts]: Actually, we should also check `peerDependencies`, too

File: dangerfile.ts [hunk 2]
```diff
@@ -37,3 +40,53 @@ if (!hasChangelog) {
...
+      const changedDependencies = _.reduce(
+        before,
+        (result, value, key) => {
+          return value === after[key] || !after[key]
+            ? result
+            : { ...result, [key]: { before: value, after: after[key] } }
+        },
+        {},
+      )
```
Comment (layershifter) [inline dangerfile.ts]: I think that we can extract this to a separate function, i.e. `getChangedDependencies(filepath)`

File: dangerfile.ts [hunk 3]
```diff
@@ -37,3 +40,53 @@ if (!hasChangelog) {
...
+
+danger.git.created_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(filepath => {
+    warn(`New package added: ${filepath}. Make sure you have OSS approval before merging!`)
+  })
+
+danger.git.modified_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(async filepath => {
```
Comment (layershifter) [inline dangerfile.ts]:
```js
danger.git.modified_files
  .filter(filepath => filepath.match(/\bpackage\.json$/))
  .reduce(async (acc, filepath) => {
    const changedDependencies = await getChangedDependencies(filepath)

    if (Object.keys(changedDependencies).length > 0) {
      acc[filename] = changedDependencies
    }

    return acc
  }, {})
  .reduce((hasWarning, changedDependencies, filename) => {
    // create markdown
    markdown(changesTabled)

    if (hasWarning) return true
    warn('Package dependencies changed. Make sure you have approval before merging!')
  }, false)
```

I expect something like this, less functions and more clear steps
/cc @kuzhelov

**Review by alinais — APPROVED** [review]

**Review by layershifter — APPROVED** [review]
Only one comment:
`hasAddedLinesAfterVersionInChangelog`, `markdownChangedDependencies`, `checkDependencyChanges` as these functions don't need any stuff from context, can we move them outside the `export default`? So we will have only the run flow in `export default`.

**Pull request merged**